docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,186 @@
+.. _amp-examples:
+
+Automatic Mixed Precision Examples
+==================================
+
+.. currentmodule:: torch.cuda.amp
+
+.. contents:: :local:
+
+.. _gradient-scaling-examples:
+
+Gradient Scaling
+^^^^^^^^^^^^^^^^
+
+The code snippets below demonstrate recommended use of :class:`torch.cuda.amp.AmpScaler`.
+The :class:`AmpScaler` instance ``scaler`` performs dynamic gradient scaling
+(maintains the scale, helps create scaled gradients to prevent
+gradient underflow, and carries out scaling-safe steps).
+
+Typical use (1 loss, 1 optimizer)
+---------------------------------
+
+::
+
+    # Create an AmpScaler instance.
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+
+        # Scale the loss, and call backward() on the scaled loss to create scaled gradients.
+        scaler.scale(loss).backward()
+
+        # Carry out a scaling-safe step.  scaler.step() unscales the optimizer's gradients
+        # and skips optimizer.step() if the gradients contain infs or NaNs.
+        scaler.step(optimizer)
+
+        # Update the scale for next iteration.
+        scaler.update()
+
+
+Scale-Aware Operations
+----------------------
+
+The gradients resulting from ``scaler.scale(loss).backward()`` are scaled.  ``scaler.step(optimizer)``
+automatically knows if it must unscale gradients before applying them.  However, networks that directly use
+gradients between the backward pass and the optimizer step should also be aware of the scale factor.
+
+Gradient clipping
+"""""""""""""""""
+
+Clipping is an operation that directly manipulates the gradients.
+:meth:`AmpScaler.unscale` may be used to unscale the gradients in-place before ``scaler.step(optimizer)``,
+allowing you to clip unscaled gradients as usual::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+
+        # Unscale the gradients owned by optimizer in-place
+        scaler.unscale(optimizer)
+
+        # Since the optimizer's owned gradients are unscaled, we can clip as usual:
+        torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm)
+
+        scaler.step(optimizer)
+        scaler.update()
+
+``scaler`` records that ``scaler.unscale(optimizer)`` has been called for this optimizer
+this iteration, so ``scaler.step(optimizer)`` knows not to redundantly unscale gradients before
+calling ``optimizer.step()``.
+
+Gradient penalty
```

Comment:
If you want to keep this then can we make it : "Working with Scaled Gradients?" Then you have "Working with Unscaled Gradients" above, and you're keeping the focus on your work.